src/Geocoder/Model/AddressFactory.php

```diff
@@ -20,6 +22,18 @@ public function createFromArray(array $results)
     {
         $addresses = [];
         foreach ($results as $result) {
+            $adminLevels = [];
+            foreach ($this->readArrayValue($result, 'adminLevels') as $level => $adminLevel) {
+                $level = $this->normalizeLevelValue($level);
+                $adminLevels[$level] = new AdminLevel(
+                    $level,
+                    $this->readStringValue($adminLevel, 'name'),
+                    $this->readStringValue($adminLevel, 'code')
+                );
+            }
+
+            ksort($adminLevels, SORT_NUMERIC);
```

Comment:
Ah cool, didn't see this test. Anyway i agree with you, this is more logical :+1: